src/Widgets/DisplayWidget.vala

```diff
@@ -26,8 +40,52 @@ public class Network.Widgets.DisplayWidget : Gtk.Grid {
     private uint cellular_animation_timeout;
     private int cellular_animation_state = 0;
 
+    private enum ConnectionState {
+        DISCONNECTED = 0,
+        CONNECTING = 1,
+        CONNECTED = 2
+    }
+
+    private ConnectionState cellular_connected = ConnectionState.DISCONNECTED;
+    private ConnectionState wifi_connected = ConnectionState.DISCONNECTED;
+    private ConnectionState wired_connected = ConnectionState.DISCONNECTED;
```

Comment:
These names are potentially confusing. I suggest `cellular_connection_state` etc.

You could then have some boolean properties e.g. bool cellular_connected { get {return cellular_connection_state != ConnectionState.DISCONNECTED;} }  to avoid using the numerical values later on.